flang/test/Lower/bbc-host-datalayout.f90

```diff
@@ -0,0 +1,9 @@
+! Test bbc set-up of the target data layout from the host.
+! RUN: bbc %s -o - | FileCheck %s
```

Comment:
Assuming there are existing tests for the flang-new driver and new ones are not required.

Reply:
That is right, flang-new changes are NFC aside from the timing when the TargetMachine is created.
This is already tested here:
- https://github.com/llvm/llvm-project/blob/820b3583c9681a1969814ac3b04a6fe41b87d079/flang/test/Driver/emit-llvm.f90#L9
- https://github.com/llvm/llvm-project/blob/820b3583c9681a1969814ac3b04a6fe41b87d079/flang/test/Driver/emit-mlir.f90#L13